common/app-starters-function-common/src/main/java/org/springframework/cloud/stream/app/function/StreamAppFunctionAutoConfiguration.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.function;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.cloud.stream.function.IntegrationFlowFunctionSupport;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.channel.AbstractMessageChannel;
+import org.springframework.integration.channel.FluxMessageChannel;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.support.ChannelInterceptor;
+
+/**
+ * Auto configuration class that provides necessary beans and configuration for adding
+ * Function composition support to the stream apps.
+ *
+ * @author Ilayaperumal Gopinathan
+ */
+@Configuration
+@ConditionalOnBean(IntegrationFlowFunctionSupport.class)
+public class StreamAppFunctionAutoConfiguration {
+
+	@Autowired(required = false)
+	private Source channel;
+
+	@Autowired
+	private IntegrationFlowFunctionSupport functionSupport;
+
+	@SuppressWarnings("resource")
+	@Bean
+	public MessageChannel processOutput() {
+		return new FluxMessageChannel();
+	}
+
+	@Bean
+	public IntegrationFlow httpProcessOutput() {
+		MessageChannel processOutput = processOutput();
+		//todo: check multiple function names and add interceptor for each function
+		//todo: this in fact needs to come from SCSt
+		QueueChannel queueChannel  = new QueueChannel();((AbstractMessageChannel) this.channel.output()).addInterceptor(new FunctionInvokerInterceptor(processOutput, queueChannel));
```

Comment:
Is adding the interceptor to invoke the function the right approach here? 

The reason behind choosing this is that there could be other interceptors configured for the Source's `output` channel and I believe they need to be completed before the composed function gets invoked. We also need to give the lowest order for this interceptor.

This also lets us *not* to make any changes into the existing app starters as they don't need to be changed to use the `Flux` message channel in order to have the functions to be invoked.

Reply:
New line here would be nice.  Also make `httpProcessOutput` `@ConditionalOnBean(Source.class)` ?

Reply:
@ilayaperumalg I may be coming late to the game, but i think you and I talked about it and as tempting as it is the interceptor strategy will not work here for many reasons.